app/src/main/java/pm/gnosis/heimdall/data/db/models/GnosisSafeDb.kt

```diff
@@ -27,6 +26,41 @@ data class GnosisSafeDb(
 fun Safe.toDb() = GnosisSafeDb(address)
 fun GnosisSafeDb.fromDb() = Safe(address)
 
+@Entity(
+    tableName = GnosisSafeOwnerDb.TABLE_NAME,
+    foreignKeys = [
+        ForeignKey(
+            entity = GnosisSafeDb::class,
+            parentColumns = [GnosisSafeDb.COL_ADDRESS],
+            childColumns = [GnosisSafeOwnerDb.COL_SAFE_ADDRESS],
+            onUpdate = ForeignKey.NO_ACTION,
+            onDelete = ForeignKey.CASCADE,
+            deferred = true
+        )
+    ],
+    indices = [
+        Index(value = [GnosisSafeOwnerDb.COL_SAFE_ADDRESS])
+    ]
+)
+data class GnosisSafeOwnerDb(
+    @PrimaryKey
+    @ColumnInfo(name = COL_PRIVATE_KEY)
+    var privateKey: EncryptedByteArray,
+
+    @ColumnInfo(name = COL_ADDRESS)
+    val address: Solidity.Address,
+
+    @ColumnInfo(name = COL_SAFE_ADDRESS)
+    val safeAddress: Solidity.Address? = null
+) {
+    companion object {
+        const val TABLE_NAME = "gnosis_safe_owners"
```

Comment:
I would prefer a table that contains safe information (GnosisSafeInfoDb). Currently that might only be the local key, but in the future we might have other information, so we could easily extend it.

Also I would use a table where we have the safe address as a primary key and the signing key as an encrypted byte array (see other comments on PR).

What do you think?

Reply:
agree. I will adjust the tables